bedrock/contentful/tests/test_contentful_api.py

```diff
@@ -466,3 +478,171 @@ def test__only_child():
+mock_list_node = {"content": [{"content": [{"value": "foo", "nodeType": "text"}], "nodeType": "list-item"}]}
+mock_hyperlink_node = {
+    "data": {"uri": "https://example.com"},
+    "content": [{"value": "Example", "nodeType": "text", "marks": []}],
+}
+
+
+def test__StrongRenderer():
+    assert StrongRenderer().render(mock_simple_node) == "<strong>foo</strong>"
+
+
+def test__EmphasisRenderer():
+    assert EmphasisRenderer().render(mock_simple_node) == "<em>foo</em>"
+
+
+@patch("bedrock.contentful.api.get_current_request")
+def test__LinkRenderer__mozilla_link(mock_get_current_request):
+
+    mock_request = Mock()
+    mock_request.page_info = {"utm_campaign": "TEST"}
+    mock_get_current_request.return_value = mock_request
+    mozilla_mock_hyperlink_node = deepcopy(mock_hyperlink_node)
+    mozilla_mock_hyperlink_node["data"]["uri"] = "https://mozilla.org/test/page/"
+    output = LinkRenderer({"text": TextRenderer}).render(mozilla_mock_hyperlink_node)
+    expected = (
+        '<a href="https://mozilla.org/test/page/?utm_source=www.mozilla.org&utm_medium=referral&utm_campaign=TEST" '
+        'data-cta-type="link" data-cta-text="Example" rel="external noopener">Example</a>'
+    )
+
+    assert output == expected
+
+
+@patch("bedrock.contentful.api.get_current_request")
+def test__LinkRenderer__mozilla_link__existing_utm(mock_get_current_request):
+
+    mock_request = Mock()
+    mock_request.page_info = {"utm_campaign": "TEST"}
+    mock_get_current_request.return_value = mock_request
+    mozilla_mock_hyperlink_node = deepcopy(mock_hyperlink_node)
+    mozilla_mock_hyperlink_node["data"]["uri"] = "https://mozilla.org/test/page/?utm_source=UTMTEST"
+    output = LinkRenderer({"text": TextRenderer}).render(mozilla_mock_hyperlink_node)
+    expected = (
+        '<a href="https://mozilla.org/test/page/?utm_source=UTMTEST" '
+        'data-cta-type="link" data-cta-text="Example" rel="external noopener">Example</a>'
+    )
+
+    assert output == expected
+
+
+def test__LinkRenderer__non_mozilla():
+    assert (
+        LinkRenderer(
+            {
+                "text": TextRenderer,
+            }
+        ).render(mock_hyperlink_node)
+        == '<a href="https://example.com" data-cta-type="link" data-cta-text="Example" rel="external noopener">Example</a>'
+    )
+
+
+def test__UlRenderer():
+    assert (
+        UlRenderer(
+            {
+                "text": TextRenderer,
+                "list-item": ListItemRenderer,
+            }
+        ).render(mock_list_node)
+        == "<ul class='mzp-u-list-styled'><li>foo</li></ul>"
+    )
+
+
+def test__OlRenderer():
+    assert (
+        OlRenderer(
+            {
+                "text": TextRenderer,
+                "list-item": ListItemRenderer,
+            }
+        ).render(mock_list_node)
+        == "<ol class='mzp-u-list-styled'><li>foo</li></ol>"
+    )
+
+
+@pytest.mark.skip("TODO")
+def test__LiRenderer():
+    assert False, "WRITE ME"
```

Comment:
⚠️

Reply:
Yep - that's getting addressed in the next batch. Thanks for the laser-eyes